src/v/cluster/rm_stm.cc

```diff
@@ -280,6 +282,10 @@ rm_stm::rm_stm(
                                model::producer_identity,
                                ss::lw_shared_ptr<ss::basic_rwlock<>>>(
       _tx_root_tracker.create_child("idempotent-producer-locks")))
+  , _tm_partition(
```

Comment:
it's better to update tx_seqs to use `tx_seq*partition_id struct instead of introducing another collection

Reply:
fixed